docs/guides/03-Styling-your-visualization-using-Expressions.md

```diff
@@ -0,0 +1,198 @@
+        width="100%"
+        height="1000"
+        style="margin: 20px auto !important"
+        frameBorder="0">
+    </iframe>
+    <a href="/developers/carto-vl/examples/#example-geometry-types">View my source code!</a>
+</div>
+
+The complete list of CARTO VL styling properties and their semantics can be seen [here](https://carto.com/developers/carto-vl/reference/#vizspec).
+
+
+### What is an expression?
+
+
+As seen above, each styling property is assigned a value that is considered to be an expression. Expressions assigned to properties that can be constants (like `red`, `#FFF`,`7`) or functions (like `sqrt`, `+`, or `ramp`).
+
+
+```CARTO_VL_Viz
+width: 7
+color: white
+```
+
+Expressions can be combined together with _expression functions_:
+
+```CARTO_VL_Viz
+width: 7+3
+color: white
+```
+
+
+*Note: about function terminology*
+
+The concept of *function* and its related terms *function call* and *function parameter* are borrowed from programming terminology.
+
+A *function* is a transformation between inputs, the *function parameters*, to an output.
+
+A *function call* is the application of the function to some values as the *function parameters*.
+
+
+Each expression imposes limitations on the types of expressions that can be used as its input parameters.
+
+
+For example, the styling below would not work because a numeric parameter (`7`) can't be added to a color parameter (`white`):
+
+```CARTO_VL_Viz
+width: 7+white
+```
+
+Similarly, even though the `7+3` is valid, this example would not work because a numeric expression (the addition) is being assigned to the styling property `color` (which accepts color expressions):
+
+```CARTO_VL_Viz
+color: 7+3
+```
+
+For a complete list of types and valid functions, see the [reference](https://carto.com/developers/carto-vl/reference/).
+
+### Color expression basics
+
+There are multiple ways to get color expressions (expressions with type color):
+- **Color constants** are colors defined by its name (`red`, `blue`, `white`...) or by its hexadecimal RGB(A) composition (#F88, #F88A, #F08080, #F08080A0).
+- **Color constructors** are functions that define colors by the components of its color space: `rgba(255, 255, 255, 0.6)`, `hsv(0, 1, 1)`.
+- The **`opacity`** function, which overrides an input color alpha channel
+- The **`ramp`** function, which will be introduced in the "Data Driven visualization" guide.
+- Other advanced expressions like color arithmetic or blending
+
+
+<div class="example-map">
+    <iframe
+        id="population-density-basic"
+        src="/developers/carto-vl/examples/maps/styling/color-spaces.html"
+        width="100%"
+        height="1000"
+        style="margin: 20px auto !important"
+        frameBorder="0">
+    </iframe>
+    <a href="/developers/carto-vl/examples#example-color-spaces">View my source code!</a>
+</div>
+
+
+You can take a look at the reference for a detailed description of all color expressions.
```

Comment:
Add a link here?

Reply:
We don't have a place with all color expressions